exasol/nb_connector/cli/processing/option_mapper.py

```diff
@@ -0,0 +1,154 @@
+
+import getpass
+import os
+from pathlib import Path
+from typing import Any
+
+from exasol.nb_connector.ai_lab_config import AILabConfig as CKey
+from exasol.nb_connector.ai_lab_config import StorageBackend
+from exasol.nb_connector.cli import reporting as report
+from exasol.nb_connector.cli.options import (
+    DOCKER_DB_OPTIONS,
+    ONPREM_OPTIONS,
+    SAAS_OPTIONS,
+)
+from exasol.nb_connector.cli.param_wrappers import ScsOption
+from exasol.nb_connector.cli.processing.backend_selector import BackendSelector
+from exasol.nb_connector.secret_store import Secrets
+
+
+class ScsCliError(Exception):
+    """
+    Indicates an error when saving or checking CLI options wrt. the Secure
+    Configuration Storage (SCS).
+    """
+
+
+SELECT_BACKEND_OPTION = ScsOption("backend", scs_key=CKey.storage_backend)
+USE_ITDE_OPTION = ScsOption("use_itde", scs_key=CKey.use_itde)
+
+
+def get_options(
+    backend: StorageBackend,
+    use_itde: bool,
+) -> list[ScsOption]:
+    def specific_options():
+        if backend == StorageBackend.saas:
+            return SAAS_OPTIONS
+        if backend != StorageBackend.onprem:
+            raise ScsCliError(f"Unsupported backend {backend}")
+        return DOCKER_DB_OPTIONS if use_itde else ONPREM_OPTIONS
+
+    return [SELECT_BACKEND_OPTION, USE_ITDE_OPTION] + specific_options()
+
+
+class OptionMapper:
+    """
+    The lifecycle of a CLI parameter comprises two phases: P1) The initial
+    definition of the parameter. P2) When the parameter is instantiated with a
+    value.
+
+    (P1) may contain a CLI name like "--db-username" while in (P2) the
+    parameter is represented as an arg name, e.g. "db_username", but also a
+    value, i.e. the name of the actual database user.
+
+    This class helps mapping both instants as P1 usually contains additional
+    information, in particular the key for the SCS fror saving the parameter's
+    value.
+
+    * Given the arg name (P2) find the (P1) instance of the parameter.
+
+    * Transfer option values as default to other options.
+
+    * Check the SCS content fore completeness.
+    """
+
+    def __init__(self, scs: Secrets, backend: StorageBackend, use_itde: bool):
+        self.scs = scs
+        self.options = get_options(backend, use_itde)
+
+    def find_option(self, arg_name: str) -> ScsOption:
+        """
+        Find the full definition of a click parameter for the specified
+        arg name.
+        """
+        try:
+            return next(o for o in self.options if o.arg_name == arg_name)
+        except StopIteration:
+            raise ScsCliError(
+                f"Couldn't find any option with parameter name {arg_name}."
+            )
+
+    def set_dynamic_defaults(self, values: dict[str, Any]) -> dict[str, Any]:
+        """
+        Some options may specify another option to get their default value
+        from, e.g. --bucketfs-host-internal reads its default value from
+        --bucketfs-host.
```

Comment:
I think the docstring is incomplete. It doesn't say what the function is doing.  Also, it's worth mentioning that it modifies the input dictionary rather than creating a new one.

Reply:
See next push, with updated docstring.

Reply:
Sorry, I might be looking at a wrong place, but I don't see any changes.

Reply:
Sorry, I updated the docstring and then accidentally deleted the change. 
Please see latest push once again, adding the sentence below.